### ISSUE TITLE
Output messages not properly formatted in IntelliJ

### ISSUE THREAD

**Issue by OGitHubUser** [rnorth/visible-assertions#1]
Messages are correctly formatted in terminal.

But in IntelliJ formatting is wrong.

<img width="872" alt="terminal" src="https://cloud.githubusercontent.com/assets/16609648/12198119/0b3bc20e-b605-11e5-9988-7b5448c39bd8.png">

<img width="1424" alt="intellij" src="https://cloud.githubusercontent.com/assets/16609648/12198130/117bc434-b605-11e5-8209-593979efaf40.png">

visible-assertion version - 1.0.4
IntelliJ version - 14.1.5

**Comment by rnorth** [comment]
Yes, you're correct, and I need to look into that. In my recent experience (more strangely!) the output is only broken when looking at the JUnit root level in the IntelliJ run window; when selecting an individual test it looks OK to me.

This suggests that there's a change needed, either to adapt to IntelliJ's behaviour, or to be more proactive about detecting ANSI colour support and disabling it in such contexts.

Thank you for raising this.

**Comment by OGitHubUser** [comment]
Strangely, every execution of individual test produce different output.

<img width="818" alt="fail1" src="https://cloud.githubusercontent.com/assets/16609648/12199690/0792d2e8-b612-11e5-8308-e81eed05c5aa.png">
<img width="693" alt="fail2" src="https://cloud.githubusercontent.com/assets/16609648/12199691/0a24ae82-b612-11e5-84dc-c8ea93ff3714.png">
<img width="714" alt="fail3" src="https://cloud.githubusercontent.com/assets/16609648/12199693/0d998290-b612-11e5-9e2c-835aff493466.png">
<img width="710" alt="success1" src="https://cloud.githubusercontent.com/assets/16609648/12199694/10ee40e8-b612-11e5-89a8-556e29a80ede.png">
<img width="629" alt="success2" src="https://cloud.githubusercontent.com/assets/16609648/12199695/14a4179e-b612-11e5-9cb7-a3e8cdd6046d.png">

Also what I noticed is that after test execution it prints correctly, but after a half a second or so, text in run console get replaced with a codes (like on a screen-shot).

**Comment by rnorth** [comment]
As a workaround for this, I've made a small change in [this branch](https://github.com/rnorth/visible-assertions/tree/intellij-ansi-workaround) to disable use of ANSI colour codes when running under IntelliJ. It's not ideal (I'd prefer to keep the coloured output), but at least it doesn't give 'corrupt' output.

It's really interesting that you noticed correct output being replaced with the escape codes, though - I'll try and reproduce.

If anybody else who comes across this knows of other tools that manage to retain coloured ANSI output in IntelliJ I'd be really keen to find out how they do it!

**Comment by OGitHubUser** [comment]
Did some more testing, and it seems like this issue only happening for messes printed before stack-trace.
Everything after stack-trace is fine. If there no stack-trace at all (all green) everything is printed fine.

Also the order of assert messages and stack-traces is mixed up. Is it at all possible to achieve in IntelliJ the same behavior as in Terminal when all assertion messages come first and then stack-traces?

<img width="1413" alt="newintellij" src="https://cloud.githubusercontent.com/assets/16609648/12232160/b4664560-b854-11e5-95ce-46b8aef0b25a.png">
<img width="795" alt="newterminal" src="https://cloud.githubusercontent.com/assets/16609648/12232176/d1b40b16-b854-11e5-9934-d19784f290cb.png">
